src/client/common/utils/async.ts

```diff
@@ -237,3 +237,25 @@ export async function flattenIterator<T>(iterator: AsyncIterator<T, void>): Prom
     }
     return results;
 }
+
+export class ChainedExecutions<T> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    private pendingCellExecution: Promise<any> = Promise.resolve();
+    /**
+     * Clears the chained promises.
+     */
+    public clear() {
+        this.pendingCellExecution = Promise.resolve();
+    }
+    public async chainExecution(next: () => Promise<T>): Promise<T> {
+        const deferred = createDeferred<T>();
+        const chainedPromise = this.pendingCellExecution.then(() => {
```

Comment:
Must use `then` to properly chain the promises.
Added `clear` to clear any previous results when starting a new batch of promises.
Calling `clear` is crucial for cases where we start a new batch and wish to ignore previous failures & results.